docs/content/guides/non-argument-deps.mdx

```diff
@@ -0,0 +1,136 @@
+---
+title: Assets without I/O | Dagster
+description: Often we don't want to build or load our assets as Python objects in memory
+---
+
+# Assets without I/O
+
+In the tutorial, all the assets we've looked at are straightforward to work with as Python objects in memory - e.g. `topstories` is created by constructing a DataFrame of the top stories from the Hacker News API.
```

Comment:
🔧 This assumes the user completed the tutorial - it feels like an odd way to introduce this guide. Is it necessary to mention the tutorial here at all?